Build failure with GHC-head 8.1.20160724

```diff
diff --git a/hackage-security/src/Hackage/Security/Client/Repository/Remote.hs b/hackage-security/src/Hackage/Security/Client/Repository/Remote.hs
index 558c982..25bbfc1 100644
--- a/hackage-security/src/Hackage/Security/Client/Repository/Remote.hs
+++ b/hackage-security/src/Hackage/Security/Client/Repository/Remote.hs
@@ -445,6 +445,12 @@ getFile cfg@RemoteConfig{..} attemptNr remoteFile method =
                                 (mustCache remoteFile)
         return (Some format, remoteTemp)
 
+    httpGetRange :: forall a. Throws SomeRemoteError
+                 => [HttpRequestHeader]
+                 -> URI
+                 -> (Int, Int)
+                 -> (HttpStatus -> [HttpResponseHeader] -> BodyReader -> IO a)
+                 -> IO a
     HttpLib{..} = cfgHttpLib
```

fixes the build.

The original error was:

```
[41 of 42] Compiling Hackage.Security.Client.Repository.Remote ( src/Hackage/Security/Client/Repository/Remote.hs, dist/dist-sandbox-dfe22761/build/Hackage/Security/Client/Repository/Remote.o )

src/Hackage/Security/Client/Repository/Remote.hs:421:13: error:
    • Couldn't match type ‘a0’ with ‘HttpStatus’
        ‘a0’ is untouchable
          inside the constraints: typ ~ Binary
          bound by the type signature for:
                     update :: typ ~ Binary =>
                               HasFormat fs f
                               -> Trusted FileInfo
                               -> Path Absolute
                               -> Int54
                               -> Verify (Some (HasFormat fs), RemoteTemp typ)
          at src/Hackage/Security/Client/Repository/Remote.hs:(397,5)-(402,58)
      Expected type: IO a0
        Actual type: IO HttpStatus
    • In a stmt of a 'do' block: return (statusCode :: HttpStatus)
      In the expression:
        do { updateServerCapabilities cfgCaps responseHeaders;
             let expectedSize = ...;
             execBodyReader targetPath expectedSize h bodyReader;
             hClose h;
             .... }
      In the second argument of ‘($)’, namely
        ‘\ statusCode responseHeaders bodyReader
           -> do { updateServerCapabilities cfgCaps responseHeaders;
                   let ...;
                   .... }’
    • Relevant bindings include
        httpGetRange :: [HttpRequestHeader]
                        -> URI
                        -> (Int, Int)
                        -> (HttpStatus -> [HttpResponseHeader] -> BodyReader -> IO a0)
                        -> IO a0
          (bound at src/Hackage/Security/Client/Repository/Remote.hs:448:5)
cabal: Error: some packages failed to install:
hackage-security-0.5.2.1 failed during the building phase. The exception was:
ExitFailure 1
```

--- a/hackage-security/src/Hackage/Security/Client/Repository/Remote.hs
+++ b/hackage-security/src/Hackage/Security/Client/Repository/Remote.hs
@@ -445,6 +445,12 @@ getFile cfg@RemoteConfig{..} attemptNr remoteFile method =
                                 (mustCache remoteFile)
         return (Some format, remoteTemp)
 
+    httpGetRange :: forall a. Throws SomeRemoteError
+                 => [HttpRequestHeader]
+                 -> URI
+                 -> (Int, Int)
+                 -> (HttpStatus -> [HttpResponseHeader] -> BodyReader -> IO a)
+                 -> IO a
     HttpLib{..} = cfgHttpLib
 
 {-------------------------------------------------------------------------------

